sonar-batch-protocol/src/main/java/org/sonar/batch/protocol/ReportStream.java

```
@@ -0,0 +1,78 @@
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.batch.protocol;
+
+import com.google.protobuf.Message;
+import com.google.protobuf.Parser;
+import org.apache.commons.io.IOUtils;
+
+import java.io.Closeable;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+public class ReportStream<R extends Message> implements Closeable, Iterable<R> {
+
+  private final Parser<R> parser;
+  private InputStream inputStream;
+
+  public ReportStream(File file, Parser<R> parser) {
+    this.parser = parser;
+    this.inputStream = ProtobufUtil.createInputStream(file);
+  }
+
+  @Override
+  public Iterator<R> iterator() {
+    return new Iterator<R>() {
+      private R currentMessage;
+
+      @Override
+      public boolean hasNext() {
+        if (currentMessage == null) {
+          currentMessage = ProtobufUtil.readInputStream(inputStream, parser);
+        }
+        return currentMessage != null;
+      }
+
+      @Override
+      public R next() {
+        if (!hasNext()) {
+          throw new NoSuchElementException();
+        }
+        R messageToReturn = currentMessage;
+        currentMessage = null;
+        return messageToReturn;
+      }
+
+      @Override
+      public void remove() {
+        throw new UnsupportedOperationException();
+      }
+    };
+  }
+
+  @Override
+  public void close() throws IOException {
+    IOUtils.closeQuietly(inputStream);
```

Do you really need to use IOUtils.closeQuietly here? You'll hide exception.